cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -266,6 +266,52 @@ void EnsureLookupTablesFilled() {}
 
 #endif  // ARROW_WITH_UTF8PROC
 
+template <typename Type>
+struct AsciiReverse : StringTransform<Type, AsciiReverse<Type>> {
+  using Base = StringTransform<Type, AsciiReverse<Type>>;
+  using offset_type = typename Base::offset_type;
+
+  bool Transform(const uint8_t* input, offset_type input_string_ncodeunits,
+                 uint8_t* output, offset_type* output_written) {
+    uint8_t utf8_char_found = 0;
```

Comment:
Use `bool` instead.

Reply:
I used a `uint8_t` because `input[i] & 0x80` generates a `0x80`. Should I add something like
`utf8_char_found |= (input[i] & 0x80 != 0);`?

Reply:
Ah, I see. No, it's ok.